### PR TITLE
`Hds::Dropdown::ListItem::Interactive` component enhancements

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

If merged, this PR will update the `Hds::Dropdown::ListItem::Interactive` component API to allow for more complex content.

### :hammer_and_wrench: Detailed description

- Adds a yielded block to the component
- Yields the `Hds::Badge` component

### :link: External links

Jira ticket: [HDS-3620](https://hashicorp.atlassian.net/browse/HDS-3620)
Figma file: [[if it applies]](https://www.figma.com/design/noyY6dUMDYjmySpHcMjhkN/branch/rdDQOAXN7HX1sj0SyO5ciW/HDS-Components?node-id=2-7&m=dev)

***

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3620]: https://hashicorp.atlassian.net/browse/HDS-3620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/components/hds/dropdown/list-item/interactive.hbs [hunk 1]
```diff
@@ -30,7 +34,11 @@
         </span>
       {{/if}}
       <Hds::Text::Body class="hds-dropdown-list-item__interactive-text" @tag="span" @size="200" @weight="medium">
-        {{this.text}}
+        {{#if (has-block)}}
+          {{yield (hash Badge=(component "hds/badge" size="small"))}}
```
Comment (alex-ju) [inline packages/components/src/components/hds/dropdown/list-item/interactive.hbs:38]: question: Is it intentional that we enforce the size here, but not [on the `@isLoading` variant](https://github.com/hashicorp/design-system/pull/2346/files#diff-6fc55f901fc63dbb36f21d330801f0ebd9b7efc61f2a2fb104893b72c9484323R13)?

File: showcase/tests/integration/components/hds/dropdown/list-item/interactive-test.js [hunk 2]
```diff
@@ -130,5 +146,24 @@ module(
...
+
+    // CONTEXTUAL COMPONENTS
+
+    test('it renders the contextual components', async function (assert) {
+      await render(
+        hbs`<Hds::Dropdown::ListItem::Interactive as |I|>
+              <I.Badge @text="Badge" />
+            </Hds::Dropdown::ListItem::Interactive>`
+      );
+      assert.dom('.hds-badge').hasText('Badge');
```
Comment (alex-ju) [inline showcase/tests/integration/components/hds/dropdown/list-item/interactive-test.js:158]: question: if we decide to set the size to small implicitly, should we check for the CSS modifier?

File: showcase/app/templates/components/dropdown.hbs [hunk 3]
```diff
@@ -458,32 +472,94 @@
...
+        </ul>
+      </div>
+    </SF.Item>
+  </Shw::Flex>
+
+  <Shw::Flex as |SF|>
+    <SF.Item @label="With long content that might push the badge down">
+      <div class="hds-dropdown__content">
+        <ul class="hds-dropdown__list">
+          <Hds::Dropdown::ListItem::Interactive @icon="settings" @trailingIcon="settings" as |I|>
```
Comment (alex-ju) [inline showcase/app/templates/components/dropdown.hbs:559]: nit: was this meant to be `external-link` as in the previous examples?
```suggestion
          <Hds::Dropdown::ListItem::Interactive @icon="settings" @trailingIcon="external-link" as |I|>
```

File: packages/components/src/components/hds/dropdown/list-item/interactive.hbs (hunk 1 above)
Comment (zamoore) [reply packages/components/src/components/hds/dropdown/list-item/interactive.hbs:38]: No, it's not. Thanks. Probably need to add some variations for that in showcase too.

File: showcase/tests/integration/components/hds/dropdown/list-item/interactive-test.js (hunk 2 above)
Comment (zamoore) [reply showcase/tests/integration/components/hds/dropdown/list-item/interactive-test.js:158]: Yes, I think that's a good idea.

File: packages/components/src/components/hds/dropdown/list-item/interactive.hbs (hunk 1 above)
Comment (alex-ju) [reply packages/components/src/components/hds/dropdown/list-item/interactive.hbs:38]: In the Dropdown Toggle Button they are always `small`, same for the Checkbox, Checkmark, and Radio equivalents. The main consideration for that is to ensure the resulting height of the ListItem is always the same (a large Badge would likely increase it). Maybe check with @LilithJames-HDS (if this hasn't already been discussed).

**Review by alex-ju — COMMENTED** [review]
We ended up with [quite a few changes](https://github.com/hashicorp/design-system/pull/2346/files) that don't belong to this PR (Table, Showcase components, etc); likely a result of rebasing it to main, where the feature branch was not rebased. Let me know if you need a hand with resolving these

**Pull request merged**